dask_ml/model_selection/_search.py

```diff
@@ -30,7 +30,11 @@
     _BaseKFold,
     _CVIterableWrapper,
 )
-from sklearn.pipeline import FeatureUnion, Pipeline
+from sklearn.pipeline import FeatureUnion
+try:
+    from imblearn.pipeline import Pipeline
+except:
+    from sklearn.pipeline import Pipeline
```

Comment:
Why is this needed? Can we always use `sklearn.pipeline.Pipeline` and check for `hasattr(x, "fit_resample")`?

Reply:
> @xKHUNx can you summarize the high-level changes? I'm a little concerned that this is affecting so many small places? For example, why does the return type of `fit_transform` need to be changed? Can that be limited to a new `fit_reshape` method that returns the transformed `yt` as well?

I assume you meant  `fit_resample`, correct me if I am wrong. My high level idea is that `yt` needs to be propagated throughout the pipeline as component that implements `fit_resample` might change the `y` values. The code works for my personal use case, where I perform RandomizedCVSearch with imblearn component in my pipeline. I can't post my code here as it is related to my work.

> Why is this needed? Can we always use `sklearn.pipeline.Pipeline` and check for `hasattr(x, "fit_resample")`?

This is needed when the refit option is selected. If I understand correctly, when the refit option is selected, dak-ml will fit the pipeline normally, without going through each compoennt in the pipeline. In this scenario, if a normal sklearn pipeline is used, it won't handle pipeline component that implements `fit_resample` instead of `fit_transform`. On the other hand, if imblearn's pipeline is used, it would behave as it should.